userspace/libsinsp/threadinfo.cpp

```diff
@@ -767,6 +768,35 @@ uint64_t sinsp_threadinfo::get_fd_limit()
 	return get_main_thread()->m_fdlimit;
 }
 
+bool sinsp_threadinfo::detect_parent_state_loop()
+{
+	// Use two pointers starting at this, traversing the parent
+	// state, at different rates. If they ever equal each other
+	// before one is NULL there's a loop.
+	sinsp_threadinfo *p1=this, *p2=this;
```

Comment:
What about naming them `turtle` and `hare` so it's clear which one goes slower/faster?